src/main/java/uk/ac/cam/cl/dtg/isaac/api/QuizFacade.java

```diff
@@ -1226,6 +1229,185 @@ public final Response getQuizAssignmentCSV(@Context final HttpServletRequest htt
+                    group.getGroupName(), new Date(), user.getGivenName(), user.getFamilyName()));
+            headerBuilder.append(",,");
+
+            List<String> quizTitles = new ArrayList<>();
+            List<String> questionTitles = new ArrayList<>();
+
+            for (QuizAssignmentDTO assignment : quizAssignments) {
+                String quizId = assignment.getQuizId();
+                IsaacQuizDTO quiz = this.quizManager.findQuiz(quizId);
+                quizTitles.add(String.format("\"%s\"", quiz.getTitle()));
+                questionTitles.addAll(getQuizQuestionTitles(quiz));
+            }
+
+            for (RegisteredUserDTO groupMember : groupMembers) {
+                UserSummaryDTO groupMemberSummary = this.userManager.convertToUserSummaryObject(groupMember);
+                List<String> row = new ArrayList<>(Arrays.asList(groupMember.getGivenName(), groupMember.getFamilyName()));
+                List<String> quizTotals = new ArrayList<>();
+                List<String> questionResults = new ArrayList<>();
+
+                for (QuizAssignmentDTO quizAssignment : quizAssignments) {
+                    String quizId = quizAssignment.getQuizId();
+                    IsaacQuizDTO quiz = this.quizManager.findQuiz(quizId);
+                    List<String> questionIds = getQuizQuestionIds(quiz);
+                    Optional<QuizUserFeedbackDTO> userFeedback = getUserFeedback(user, quizAssignment, quiz, groupMembers).stream()
+                            .filter(f -> f.getUser().getId().equals(groupMember.getId())).findFirst();
+                    if (!userFeedback.isPresent()) {
+                        // This looks like it should work but I can't test it as I don't know how to retrieve a
+                        // non-present user feedback. If everything is set up correctly with retrieving user
+                        // feedback, this branch should never happen -- so maybe we might as well forcibly unwrap
+                        // and deal with the exception because something has gone very wrong somewhere anyway.
+                        quizTotals.add(""); // We should probably write something here to make debugging easier...
+                        long questionCount = quiz.getChildren().stream()
+                                .filter(c -> c instanceof IsaacQuizSectionDTO)
+                                .map(s -> ((IsaacQuizSectionDTO) s).getChildren().stream().filter(t -> t instanceof IsaacQuestionBaseDTO).count())
+                                .reduce(0L, Long::sum);
+                        for (long i = 0L; i < questionCount; ++i) {
+                            questionResults.add("-");
+                        }
+                    } else {
+                        QuizFeedbackDTO.Mark overallMark = Objects.requireNonNull(userFeedback.get().getFeedback()).getOverallMark();
```

Comment:
If one or more people aren't sharing their data in the assigned quizzes for that group then this throws a null pointer exception and doesn't work!